packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,258 @@
+
+    if (existsSync(npxCacheLocation)) {
+      await fsp.rm(npxCacheLocation, { recursive: true });
+    }
+  } else if (packageManager.startsWith('yarn')) {
+    if (packageManager === 'yarn-modern') {
+      await execa('corepack', ['enable'], execaOptions);
+      await execa('yarn', ['init', '-2'], execaOptions);
+
+      await execa(
+        'yarn',
+        ['config', 'set', 'npmRegistryServer', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa(
+        'yarn',
+        ['config', 'set', 'unsafeHttpWhitelist', 'localhost'],
+        execaOptions
+      );
+      await execa(
+        'yarn',
+        ['config', 'set', 'nodeLinker', 'node-modules'],
+        execaOptions
+      );
+    } else {
+      await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+      await execa(
+        'yarn',
+        ['config', 'set', 'registry', 'http://localhost:4873'],
+        execaOptions
+      );
+      await execa('yarn', ['config', 'get', 'registry'], execaOptions);
+    }
+    await execa('yarn', ['cache', 'clean'], execaOptions);
+  } else if (packageManager === 'pnpm') {
+    await execa('npm', ['install', '-g', packageManager], {
+      stdio: 'inherit',
+    });
+    await execa(packageManager, ['--version']);
+    await execa(packageManager, [
+      'config',
+      'set',
+      'registry',
+      'http://localhost:4873',
+    ]);
+    await execa(packageManager, ['config', 'get', 'registry']);
+  }
+
+  return { packageManagerExecutable, packageManager };
+};
+
+void describe('installs expected packages and scaffolds expected files', async () => {
+  let branchBackendIdentifier: BackendIdentifier;
+  let testBranch: TestBranch;
+  let cfnClient: CloudFormationClient;
+  const tempDir = await fsp.mkdtemp(
+    path.join(os.tmpdir(), 'test-create-amplify')
+  );
+  const { packageManagerExecutable, packageManager } =
+    await setupPackageManager(tempDir);
+
+  before(async () => {
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
+    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+
+    cfnClient = new CloudFormationClient(e2eToolingClientConfig);
+    testBranch = await amplifyAppPool.createTestBranch();
+    branchBackendIdentifier = {
+      namespace: testBranch.appId,
+      name: testBranch.branchName,
+      type: 'branch',
+    };
+  });
+
+  after(async () => {
+    await cfnClient.send(
+      new DeleteStackCommand({
+        StackName: BackendIdentifierConversions.toStackName(
+          branchBackendIdentifier
+        ),
+      })
+    );
+    await fsp.rm(tempDir, { recursive: true });
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  void it(`starting project`, async () => {
+    await execa(
+      packageManager.startsWith('yarn') ? 'yarn' : packageManager,
```

Comment:
why don't we use `packageManagerExecutable` from above ?

Reply:
because for `npm`, we don't want to use `npx`

Reply:
Done 222249a973 <- See if that's what you meant?